### ISSUE TITLE
Admin console config doesn't apply `when` to `groups` as documented

### ISSUE THREAD

**Issue by Iristyle** [replicatedhq/kots#659]
Documentation states that `when` applies to `groups` at https://kots.io/reference/v1beta1/config/#when

However, I can't seem to make it hide a group element in the config ui, using something like this:

```
apiVersion: kots.io/v1beta1
kind: Config
metadata:
  name: myconfig
spec:
  groups:
  - name: application_setup
    title: Application Setup
    when: "false"
```

I've tried several variations for the when clause, including a `{{repl ... }}` that returns false, and none seem to take effect.

**Comment by dexhorthy** [comment]
cc @marccampbell  -- @Iristyle this is solved by this PR, but we're working through backwards-compat issues https://github.com/replicatedhq/kots/pull/326

**Comment by marccampbell** [comment]
Fixed in 1.17 (release is mid july)
